File: 1044-find-common-characters/find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
        int alphabets[26];
        for(char c:words[0]) alphabets[c-'a']++;

        for(int i=1;i<words.size();++i){
            int alpha[26]={};
            for(char c:words[i]){
                alpha[c-'a']++;
            }
            for(int j=0;j<26;++j) alphabets[j]=min(alphabets[j],alpha[j]);
        }
        vector<string> result;
        for(int i=0;i<26;++i){
            while(alphabets[i]>0){
                char c = i+'a';
                string s; s.push_back(c);
                result.push_back(s);
                --alphabets[i];
            }
        }
        return result;
    }
};

File: 1044-find-common-characters/find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        counter = [0 for i in range(26)]
        for char in words[0]:
            idx = ord(char)-ord("a")
            counter[idx]+=1
        
        for i in range(1,len(words)):
            second_counter = [0 for i in range(26)]
            for char in words[i]:
                idx = ord(char)-ord("a")
                second_counter[idx]+=1
            
            for j in range(26):
                counter[j] = min(counter[j],second_counter[j])
        
        sol =[]
        for idx,num in enumerate(counter):
            for _ in range(num):
                sol.append(chr(idx+ord("a")))
        
        return sol